app/routes/goal_routes.py

```diff
@@ -0,0 +1,105 @@
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+from .helper import validate_record
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+# CREATE goal
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.create(request_body)
+    except KeyError:
+        return abort(make_response(jsonify({"details":"Invalid data"}), 400))
+        
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = {}
+    response_body['goal'] = new_goal.to_json()
+
+    return jsonify(response_body), 201
+
+
+# GET ALL goals
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_json() for goal in goals]
+
+    return jsonify(goals_response), 200
+
+# GET one goal
+@goals_bp.route("/<goal_id>", methods=["GET"])  
+def read_one_goal(goal_id):
+    goal = validate_record(Goal, goal_id)
+    return jsonify({
+        "goal": goal.to_json()
+        })
+
+
+# UPDATE one goal
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_record(Goal, goal_id)
+    request_body = request.get_json()
+
+    try:
+        goal.title = request_body["title"]
+    except KeyError:
+        return abort(make_response(jsonify({"details":"Invalid data"}), 400))
+
+    db.session.commit()
+
+    response_body = {}
+    response_body['goal'] = goal.to_json()
+
+    return jsonify(response_body), 200
```

Comment:
we could go a step further and combine all three of these lines:

```suggestion
    return jsonify({"goal": goal.to_json()}), 200
```

This works best when the response body is simple and only takes up a line. If it got bigger, with several keys and values, then assigning it to a variable first, then passing it in would be the way to go

Reply:
I like it. Thank you!